pkg/apis/provisioning/v1alpha5/requirements.go

```diff
@@ -203,6 +140,7 @@ func (r Requirements) CapacityTypes() stringsets.String {
 }
 
 // Validate validates the feasibility of the requirements.
+// Do not apply validation to requiremnts after merging with other requirements.
```

Comment:
Why not?

Reply:
After merging requirements, values may got canceled out and validation will return error due to 
```
if r.Get(key).Len() == 0 && !r.hasRequirement(withKeyAndOperator(key, v1.NodeSelectorOpDoesNotExist)) {
	errs = multierr.Append(errs, fmt.Errorf("no feasible value for key %s", key))
}
```
People should really use compatible in this case.